packages/desktop-client/src/components/budget/BalanceWithCarryover.tsx

```diff
@@ -54,37 +67,37 @@ export function BalanceWithCarryover({
           )
         }
         style={{
+          overflow: 'hidden',
+          textOverflow: 'ellipsis',
           textAlign: 'right',
           ...(!disabled && {
             cursor: 'pointer',
           }),
           ...props.style,
         }}
       />
-      {carryoverValue && (
-        <View
-          style={{
-            alignSelf: 'center',
-            marginLeft: 2,
-            position: 'absolute',
-            right: isNarrowWidth ? '-8px' : '-4px',
-            top: 0,
-            bottom: 0,
-            justifyContent: 'center',
-            ...carryoverStyle,
-          }}
-        >
-          <SvgArrowThinRight
-            width={carryoverStyle?.width || 7}
-            height={carryoverStyle?.height || 7}
-            style={makeBalanceAmountStyle(
-              balanceValue,
-              goalValue,
-              budgetedValue,
-            )}
-          />
-        </View>
-      )}
-    </>
+      {carryoverValue &&
+        (carryoverView ? (
+          carryoverView(valueStyle?.color)
```

Comment:
Instead of passing the color, just pass in the carryover amount so that the caller have the flexibility to style accordingly

Reply:
I'm not really too concerned about the CPU time but we need that flexibility since some code apply some modifications to the styling e.g. balance modals. Passing the value style also works for me.